acm/src/main/java/org/literacybridge/acm/store/AudioItemIndex.java

```diff
@@ -380,6 +361,24 @@ private Playlist getPlaylist(final String uuid, final Playlist playlist) throws
         }
     }
 
+    private void loadPlaylistFromPostingList(LeafReader leafReader, String playlistUuid, Playlist.Builder playlistBuilder) throws IOException {
+        PostingsEnum tp = leafReader.postings(new Term(PLAYLISTS_FIELD, playlistUuid), PostingsEnum.PAYLOADS);
+        if (tp != null) {
+            while (tp.nextDoc() != DocIdSetIterator.NO_MORE_DOCS) {
+                // important: Lucene applies deletes of documents to posting lists lazily when it
+                // performs a segment merge, so it is necessary here to check if this audioItem is deleted
+                if (leafReader.getLiveDocs() == null || leafReader.getLiveDocs().get(tp.docID())) {
+                    // TODO: we could also use the termPosition instead of a payload to store the playlist position
```

Comment:
Thanks!

I updated master with a bump to the version number, just FYI, it won't cause any trouble.

When you are ready to check this in, please rebase onto master and push that. Then delete the old branch; this change is fully self-contained in its own commit, so the branch has no historical value. 

Cheers,

-bill

On Apr 28, 2016, at 5:04 PM, Michael Busch <notifications@github.com> wrote:

In acm/src/main/java/org/literacybridge/acm/store/AudioItemIndex.java <https://github.com/LiteracyBridge/acm/pull/6#discussion_r61530741>:

> @@ -380,6 +361,24 @@ private Playlist getPlaylist(final String uuid, final Playlist playlist) throws
>          }
>      }
>  
> +    private void loadPlaylistFromPostingList(LeafReader leafReader, String playlistUuid, Playlist.Builder playlistBuilder) throws IOException {
> +        PostingsEnum tp = leafReader.postings(new Term(PLAYLISTS_FIELD, playlistUuid), PostingsEnum.PAYLOADS);
> +        if (tp != null) {
> +            while (tp.nextDoc() != DocIdSetIterator.NO_MORE_DOCS) {
> +                // important: Lucene applies deletes of documents to posting lists lazily when it
> +                // performs a segment merge, so it is necessary here to check if this audioItem is deleted
> +                if (leafReader.getLiveDocs() == null || leafReader.getLiveDocs().get(tp.docID())) {
> +                    // TODO: we could also use the termPosition instead of a payload to store the playlist position
Ok fair enough, I removed the comment.

—
You are receiving this because you were assigned.
Reply to this email directly or view it on GitHub <https://github.com/LiteracyBridge/acm/pull/6/files/e06564134f3a9346effa5d7aa6125f0f70fdcf1b#r61530741>